source/nodejs/adaptivecards-templating/src/template-engine.ts

```diff
@@ -531,9 +532,22 @@ export class Template {
      * @returns A value representing the expanded template. The type of that value
      *   is dependent on the type of the original template payload passed to the constructor.
      */
-    expand(context: IEvaluationContext): any {
+    expand(context: IEvaluationContext, errorLog?: String[]): any {
```

Comment:
errorLog is a hacky way and I would not use it in production. I have two alternatives:
1. add a feature flag to Template. For example,  Template.treatInvalidWhenAsFalse. and throw exception if user set this flag to false (For example, used by designer) to report error. 3rd party use this flag to switch back to old logic too if don't want this feature. 
2. 
```
expand(context: IEvaluationContext,  errorHandler?: function(error, errorDetail) => bool): any
```

dev may handle the error like this:

```
expand(context: IEvaluationContext,  (error, detail) => {  console.log(detail); return false} );
```

Reply:
I agree that this felt hacky which is why I made this a draft.

From my understanding, I was supposed to be making a more generic error handling system that we could use in future scenarios. Should I be catering to the specific boolean case instead?

Reply:
I did simple research on JS implementation. maybe it's easy to add new stuff to IEvaluationContext or Template to get the support the error log.

```
export interface IEvaluationContext {
    /**
     * The root data object the template will bind to. Expressions that refer to $root in the template payload
     * map to this field. Initially, $data also maps to $root.
     */
    $root: any,
    logEvaluationError? : (error) => void,
    featureflag: true,
}

```

or 
```
class Template {
...
    expand(context: IEvaluationContext): any;
    getEvaluationError(): string;
}
```

Reply:
I like the idea of collecting the errors in the template object so that developers can get them after expanding.

@jwoo-msft what do you think?

Reply:
@anna-dingler that makes sense to me; AdaptiveCards parser works that way, too.

I think adding it to the Template instance makes more sense to me since EvaluationContext's role is more specific.

Reply:
@licanhua @jwoo-msft  Thanks for the help here! Just updated it - let me know what you all think!